### PR TITLE
db: Remove EventTag relation to TagKey in deletions (to drop unncessary index)

### PR DESCRIPTION
**Pull request by bretthoerner**
As discussed, deletes will be handled by `Group` and `Event`:

https://github.com/getsentry/sentry/blob/13ab6682a4fbe2b14f1d33e9974c6eee9265e333/src/sentry/tagstore/legacy/backend.py#L46-L50

https://github.com/getsentry/sentry/blob/13ab6682a4fbe2b14f1d33e9974c6eee9265e333/src/sentry/tagstore/legacy/backend.py#L57-L61

### CODE REVIEW

**Comment by bretthoerner** [comment]
Oh, I'll split the migration out to its own PR so you can deploy this and test `INVALID`.

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR includes migrations</td>
    </tr>
  </tbody>
</table>

## Migration Checklist

- [ ] new columns need to be nullable (unless table is new)
- [ ] migration with any new index needs to be done concurrently
- [ ] data migrations should not be done inside a transaction
- [ ] before merging, check to make sure there aren't conflicting migration ids


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Review by mattrobenolt — CHANGES_REQUESTED** [review]
Let’s split the model change and the migration to another PR just so we can verify first.

File: src/sentry/tagstore/legacy/models/eventtag.py [hunk 1]
```diff
@@ -30,8 +30,6 @@ class Meta:
         app_label = 'sentry'
         db_table = 'sentry_eventtag'
         unique_together = (('event_id', 'key_id', 'value_id'), )
-        index_together = (
-            ('project_id', 'key_id', 'value_id'), ('group_id', 'key_id', 'value_id'),
-        )
+        index_together = (('group_id', 'key_id', 'value_id'), )
```
Comment (mattrobenolt) [inline src/sentry/tagstore/legacy/models/eventtag.py]: Revert this bit too just for now.

**Review by mattrobenolt — APPROVED** [review]
🥇

**Pull request merged**